Enterprise-CMCS/macpro-mdct-qmr
# Consolidate CommonQuestions - WhyAreYouNotReporting

Deployed env: https://d3533jq1kcfhva.cloudfront.net/

### Description

This PR combines the WhyAreYouNotReporting question. Everything was mostly the same except that in 2023 & 2024, we added a flag to toggle the covid checkbox option. 

AC:
- Remove interface WhyAreYouNotReporting from each years types folder and add it to a new TypeWhyAreYouNotReporting.tsx in shared/types folder. Relink those files to centralized the types
- Add a new WhyAreYouNotReporting.tsx to "shared/commonQuestions folder". This is where you will combined the components from each year into one.
- Pull the labels from the combined components to the "labels/{year}/commonQuestions.tsx", make sure the text matches what it looks like before consolidation.
- Update the links in each measures CommonQuestions/index.ts from "./WhyAreYouNotReporting" to "shared/commonQuestions/WhyAreYouNotReporting"
- Copy & Update unit test for 2024 as WhyAreYouNotReporting.test.tsx in "shared/commonQuestions folder"
- Clean up by deleting WhyAreYouNotReporting folder from each years CommonQuestions folder.
- Fix any linking issues that pops up from deleting the files.

### Related ticket(s)


CMDCT-3353

---
### How to test

1) Sign into QMR, any state user
2) Select any adult measure in reporting year 2024, e.g. AAB-AD 
3) For "Are you reporting on this measure?" section, select No. This will open the Why Are You Not Reporting section.
 
![Screenshot 2024-04-17 at 3 12 57 PM](https://github.com/Enterprise-CMCS/macpro-mdct-qmr/assets/127151429/c41d7ae2-f35c-4002-ac2a-5de174fb90c3)

4) Click all the checkboxes and sub inputs. There should be no erroring of the page.
5) Repeat it 2021, 2022 & 2023.

### Notes




---
### Pre-review checklist
